src/pymor/core/cache.py

```diff
@@ -182,6 +182,10 @@ def get(self, key):
         return has_key, self._cache.get(key, default=None)
 
     def set(self, key, value):
+        from pymor.vectorarrays.mpi import MPIVectorArray
+        if any(isinstance(v, MPIVectorArray) for v in value):
+            getLogger('pymor.core.cache.DiskRegion').warning('Disk based caching of MPI objects is not supported.')
+            return
```

Comment:
Unfortunately this fails in CI because not every cached `value` is an iterable. It's also probably better not to iterate over every `value` (could be a very long NumPy array). Maybe it is enough, to explicitly check for tuples/lists/dicts?

Reply:
With `LTIModel`s the cached value is a tuple of the actual cached solution and the number of default changes. I don't get what you are caching, so it might be best to switch to your first suggestion. (Still everything red, I don't know what is going on there.)

In `mpi_wrap_model` with `use_with=true` `with_` is called that, according to the documentation, resets the `cache_region` to the class default.
As no class with default `cache_region='disk'` exists (so far), it does not really make sense to check the `cache_region`.

Reply:
You're right that in practice the cached value is always a pair of the actual value and `defaults_changes()`. However, that is not part of the `CacheRegion` api, and the test suite also passes non-iterable values.

I think I would still prefer to handle this in the `DiskRegion`. What about the following approach: We implement `MPIVectorArray.__getstate__` to always raise an execption to ensure that `MPIVectorArrays` cannot be pickled. Then we catch the corresponding exception in `DiskRegion.set` and convert it to a meaningful warning? Just disabling the disk-based caching in `mpi_wrap_model` has the disadvantage that it does not cover other cases where an `MPIVectorArray` might end up in a disk-based cache. Also, checking for `region == 'disk'` is generally insufficient as `pymor.core.cache_regions` is, in principle, user configurable.

Regarding the current CI errors: these are clearly unrelated to your PR. I will look into them later today.